x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -198,13 +206,72 @@ export class ManifestManager {
     );
   }
 
+  /**
+   * Builds an artifact (by policy) based on the current state of the
+   * artifacts list (Trusted Apps, Host Iso. Exceptions, Event Filters, Blocklists)
+   * (which uses the `exception-list-agnostic` SO type)
+   */
+  protected async buildArtifactsByPolicy(
+    allPolicyIds: string[],
+    supportedOSs: string[],
+    osOptions: BuildArtifactsForOsOptions
+  ) {
+    const policySpecificArtifacts: Record<string, InternalArtifactCompleteSchema[]> = {};
+    await pMap(
+      allPolicyIds,
+      async (policyId) => {
+        for (const os of supportedOSs) {
+          policySpecificArtifacts[policyId] = policySpecificArtifacts[policyId] || [];
+          policySpecificArtifacts[policyId].push(
+            await this.buildArtifactsForOs({ os, policyId, ...osOptions })
+          );
+        }
+      },
+      {
+        concurrency: 5,
+        /** When set to false, instead of stopping when a promise rejects, it will wait for all the promises to
+         * settle and then reject with an aggregated error containing all the errors from the rejected promises. */
+        stopOnError: false,
```

Comment:
I think errors are not handled here - they should be handled by checking the return value of the function `pMap`, if I understand it correctly. what do you think?

Reply:
`pMap` does not returns anything (just an array of `Promise<void[]>`). What this `stopOnError: false` does is, instead of throwing at the first error and stop from running the other iterations, it runs all the iterations and then throws with all the errors aggregated.

Reply:
oh, I see, thanks for the explanation!